tiledb/sm/c_api/tiledb.cc

```diff
@@ -4961,6 +4962,23 @@ int32_t tiledb_fragment_info_get_fragment_name(
   return TILEDB_OK;
 }
 
+int32_t tiledb_fragment_info_get_fragment_name_v2(
+    tiledb_ctx_t* ctx,
+    tiledb_fragment_info_t* fragment_info,
+    uint32_t fid,
+    tiledb_string_t** name) {
+  if (sanity_check(ctx) == TILEDB_ERR ||
+      sanity_check(ctx, fragment_info) == TILEDB_ERR)
+    return TILEDB_ERR;
+
+  const std::string& nameStr =
+      fragment_info->fragment_info_->get_fragment_name_str(fid);
+
+  *name = tiledb_string_handle_t::make_handle(nameStr);
```

Comment:
I guess this is not the correct way to create the handle.

Reply:
Yes. Usage is illustrated in the unit tests in `tiledb/api/c_api/string/test/unit_capi_string.cc`.